src/main/scala/zio/kafka/admin/AdminClient.scala

```diff
@@ -875,14 +876,18 @@ object AdminClient extends Accessible[AdminClient] {
   final case class OffsetAndMetadata(
     offset: Long,
     leaderEpoch: Option[Int] = None,
-    metadata: Option[String] = None
+    metadata: Option[String] = None //non empty string
```

Comment:
I am not sure what you don't like.
Is it:
- the comment saying it shouldn't be an empty string? I don't really like it nor I would rather go with NonEmptyString, but I don't know if bringing zio.prelude as a dependency is a good idea.
 - the fact that I transform "" to None when going from java model to scala?

Reply:
1/ yes
2/ yes, but also that this does not remove the fact that from a caller's point of view, and a type perspective, the caller must still deal with Some("") if that's important for its business.

Reply:
I understand, but from a type perspective, the caller would still need to do `.filter(_.isNotEmpty)` if it really matters for its business.

More importantly, what's the use case? Correct me if I'm wrong but we don't use them anywhere? (pretty sure I never saw them used in the Kafka world). What does it bring to test something we don't use? Should we not remove metadata from our tests/assertions (because serving no business)?

Reply:
I agree that the PR is not solving the problem fully and that I needed that only for equality purpose for testing, nothing related to business, I close it.